src/core/action-creator.ts

```diff
@@ -7,262 +7,266 @@ import Adapters from './adapters';
+        (dispatch) => {
+          const query = search.query && search.query.trim();
+          if (query || query === null) {
+            dispatch(action(Actions.UPDATE_SEARCH, { ...<any>search, query }, metadata));
+          }
+        },
+
+      search: (query: string = Selectors.query(flux.store.getState())) =>
+        actions.updateSearch({ query, clear: true }),
+
+      resetRecall: (query: string = null, { field: navigationId, index }: { field: string, index: number } = <any>{}) =>
+        actions.updateSearch({ query, navigationId, index, clear: true }),
+
+      resetQuery: () => actions.updateSearch({ query: null }),
+
+      selectRefinement: (navigationId: string, index: number): Actions.Thunk<Actions.SelectRefinement> =>
+        conditional((state) =>
+          Selectors.isRefinementDeselected(state, navigationId, index),
+          Actions.SELECT_REFINEMENT, { navigationId, index }, metadata),
+
+      deselectRefinement: (navigationId: string, index: number): Actions.Thunk<Actions.DeselectRefinement> =>
+        conditional((state) =>
+          Selectors.isRefinementSelected(state, navigationId, index),
+          Actions.DESELECT_REFINEMENT, { navigationId, index }, metadata),
+
+      selectCollection: (id: string): Actions.Thunk<Actions.SelectCollection> =>
+        conditional((state) =>
+          state.data.collections.selected !== id,
+          Actions.SELECT_COLLECTION, id, metadata),
+
+      selectSort: (index: number): Actions.Thunk<Actions.SelectSort> =>
+        conditional((state) =>
+          state.data.sorts.selected !== index,
+          Actions.SELECT_SORT, index, metadata),
+
+      updatePageSize: (size: number): Actions.Thunk<Actions.UpdatePageSize> =>
+        conditional((state) =>
+          Selectors.pageSize(state) !== size,
+          Actions.UPDATE_PAGE_SIZE, size, metadata),
+
+      updateCurrentPage: (page: number): Actions.Thunk<Actions.UpdateCurrentPage> =>
+        conditional((state) =>
+          page !== null && state.data.page.current !== page,
+          Actions.UPDATE_CURRENT_PAGE, page, metadata),
+
+      updateDetailsId: (id: string): Actions.Thunk<Actions.UpdateDetailsId> =>
+        thunk(Actions.UPDATE_DETAILS_ID, id, metadata),
+
+      updateAutocompleteQuery: (query: string): Actions.Thunk<Actions.UpdateAutocompleteQuery> =>
+        conditional((state) =>
+          state.data.autocomplete.query !== query,
+          Actions.UPDATE_AUTOCOMPLETE_QUERY, query, metadata),
+
+      // response action creators
+      receiveSearchResponse: (results: Results): Actions.Thunk<any> =>
+        (dispatch, getState) => {
+          const updates = [];
+          const state = getState();
+          if (results.redirect) {
+            updates.push(dispatch(actions.receiveRedirect(results.redirect)));
+          }
+          const recordCount = Adapters.Search.extractRecordCount(results);
+          updates.push(
```

Comment:
Should we always send all of these actions? For example, should we send a receiveNavigation action when a user updates the page size (navigations haven't changed)?